lib/bolt_spec/run.rb

```diff
@@ -130,20 +130,31 @@ class BoltRunner
       # still be loaded
       def self.with_runner(config_data, inventory_data)
         Dir.mktmpdir do |boltdir_path|
-          config = Bolt::Config.new(Bolt::Boltdir.new(boltdir_path), config_data || {})
-          inventory = Bolt::Inventory.new(inventory_data || {}, config)
-          yield new(config, inventory)
+          runner = new(config_data, inventory_data, boltdir_path)
+          # runner.setup
+          yield runner
         end
       end
 
-      attr_reader :config, :inventory
-
-      def initialize(config, inventory)
-        @config = config
-        @inventory = inventory
+      def initialize(config_data, inventory_data, boltdir_path)
+        @config_data = config_data || {}
+        @inventory_data = inventory_data || {}
+        @boltdir_path = boltdir_path
         @analytics = Bolt::Analytics::NoopClient.new
       end
 
+      def config
+        @config = Bolt::Config.new(Bolt::Boltdir.new(@boltdir_path), @config_data)
```

Comment:
Does this need to be recreated every time?

Reply:
no that was a typo.